gomock/call.go

```diff
@@ -225,15 +225,17 @@ func (c *Call) SetArg(n int, value interface{}) *Call {
 	c.t.Helper()
 
 	mt := c.methodType
-	// TODO: This will break on variadic methods.
-	// We will need to check those at invocation time.
-	if n < 0 || n >= mt.NumIn() {
+	if (n < 0 || n >= mt.NumIn()) && !mt.IsVariadic() {
```

Comment:
Could you please add some tests for this. You can either do a traditional tests or a sample code generation like others in the `mockgen/internal/test` dir.